traffic_ops/traffic_ops_golang/api/generic_crud.go

```diff
@@ -80,8 +81,33 @@ func GenericCreate(val GenericCreator) (error, error, int) {
 	} else if rowsAffected > 1 {
 		return nil, errors.New("too many ids returned from " + val.GetType() + " insert"), http.StatusInternalServerError
 	}
+
 	val.SetKeys(map[string]interface{}{"id": id})
 	val.SetLastUpdated(lastUpdated)
+
+	if r, ok := val.(Reader); ok {
+		return ReadBack(r)
+	}
+
+	return nil, nil, http.StatusOK
+}
+
+func ReadBack(val Reader) (error, error, int) {
+
+	i := val.(Identifier)
+	keys, _ := i.GetKeys()
+	inf := val.APIInfo()
+	inf.Params["id"] = strconv.Itoa(keys["id"].(int))
```

Comment:
This'll panic if the object doesn't have an `id`, which sort of defeats the purpose of `APIInfo.IntParams` and `Reader.GetKeys`.